arm/compute/client.go

```diff
@@ -0,0 +1,2031 @@
+		"publisherName":  url.QueryEscape(publisherName),
+		"subscriptionId": url.QueryEscape(client.SubscriptionId),
+	}
+
+	queryParameters := map[string]interface{}{
+		"api-version": ApiVersion,
+	}
+
+	return autorest.DecoratePreparer(
+		client.ListOffersRequestPreparer(),
+		autorest.WithPathParameters(pathParameters),
+		autorest.WithQueryParameters(queryParameters)).Prepare(&http.Request{})
+}
+
+func (client *VirtualMachineImagesClient) ListOffersRequestPreparer() autorest.Preparer {
+	return autorest.CreatePreparer(
+		autorest.AsJSON(),
+		autorest.AsGet(),
+		autorest.WithBaseURL(client.BaseUri),
+		autorest.WithPath("/subscriptions/{subscriptionId}/providers/Microsoft.Compute/locations/{location}/publishers/{publisherName}/artifacttypes/vmimage/offers"))
+}
+
+// ListPublishers gets a list of virtual machine image publishers.;
+//
+func (client *VirtualMachineImagesClient) ListPublishers(location string) (result []VirtualMachineImageResource, err error) {
+
+	req, err := client.NewListPublishersRequest(location)
+	if err != nil {
+		return result, fmt.Errorf("compute: Failure preparing VirtualMachineImagesClient.ListPublishers request (%v)", err)
+	}
+
+	req, err = autorest.Prepare(
+		req,
+		client.WithAuthorization(),
+		client.WithInspection())
+	if err != nil {
+		return result, fmt.Errorf("compute: Failure sending VirtualMachineImagesClient.ListPublishers request (%v)", err)
+	}
+
+	resp, err := autorest.SendWithSender(client, req)
+	if err != nil {
+		return result, fmt.Errorf("compute: Failure sending VirtualMachineImagesClient.ListPublishers request (%v)", err)
+	}
+
+	result = []VirtualMachineImageResource{}
+	err = autorest.Respond(
+		resp,
+		client.ByInspecting(),
+		autorest.WithErrorUnlessOK(),
+		autorest.ByUnmarshallingJSON(&result),
+		autorest.ByClosing())
+
+	return result, err
+}
+
+func (client *VirtualMachineImagesClient) NewListPublishersRequest(location string) (*http.Request, error) {
+
+	pathParameters := map[string]interface{}{
+		"location":       url.QueryEscape(location),
+		"subscriptionId": url.QueryEscape(client.SubscriptionId),
+	}
+
+	queryParameters := map[string]interface{}{
+		"api-version": ApiVersion,
+	}
+
+	return autorest.DecoratePreparer(
+		client.ListPublishersRequestPreparer(),
+		autorest.WithPathParameters(pathParameters),
+		autorest.WithQueryParameters(queryParameters)).Prepare(&http.Request{})
+}
+
+func (client *VirtualMachineImagesClient) ListPublishersRequestPreparer() autorest.Preparer {
+	return autorest.CreatePreparer(
+		autorest.AsJSON(),
+		autorest.AsGet(),
+		autorest.WithBaseURL(client.BaseUri),
+		autorest.WithPath("/subscriptions/{subscriptionId}/providers/Microsoft.Compute/locations/{location}/publishers"))
+}
+
+// ListSkus gets a list of virtual machine image skus.;
+//
+func (client *VirtualMachineImagesClient) ListSkus(location string, publisherName string, offer string) (result []VirtualMachineImageResource, err error) {
```

Comment:
@brendandixon golint usually complains if you use for instance `Url` instead of `URL` or `SomeId` instead of `SomeID`. It has a dictionary I believe and the reason it's not failing on this PR is because `.travis.yml` does not include this package. (if you update .travis.yml, you'll probably encounter many errors).

To make the public facing type/field names Go-idiomatic, I went through the entire codebase and fixed all such  mixed-case abbreviations manually... Otherwise golint will just complain.